py3status/modules/sysdata.py

```diff
@@ -126,6 +132,49 @@ def memory(self, unit='GiB'):
         # Otherwise, results are in gigabytes.
         return total_mem, used_mem, used_mem_p, unit
 
+    def swap(self, unit='GiB'):
+        """
+        Parse /proc/meminfo, grab the swap capacity and used size
+        then return; Swap size 'total_swap', Used_swap, percentage
+        of used spwap, and units of swap (KiB, MiB, GiB).
+        """
+
+        memi = {}
+        with open('/proc/meminfo', 'r') as fd:
+            for s in fd:
+                tok = s.split()
+                memi[tok[0]] = float(tok[1])
+
+        try:
+            total_swap_kib = memi["SwapTotal:"]
+            used_swap_kib = (total_swap_kib -
+                            memi["SwapFree:"])
+            used_swap_p = 100 * used_swap_kib / total_swap_kib
+            multiplier = {
+                'KiB': ONE_KIB / ONE_KIB,
+                'MiB': ONE_KIB / ONE_MIB,
+                'GiB': ONE_KIB / ONE_GIB,
+            }
+            if unit.lower() == 'dynamic':
+                # If less than 1 GiB, use MiB
+                if (multiplier['GiB'] * total_swap_kib) < 1:
+                    unit = 'MiB'
+                else:
+                    unit = 'GiB'
+            if unit in multiplier.keys():
+                total_swap = multiplier[unit] * total_swap_kib
+                used_swap = multiplier[unit] * used_swap_kib
+            else:
+                raise ValueError(
+                    'unit [{0}] must be one of: KiB, MiB, GiB, dynamic.'.format(unit))
```

Comment:
Again this code is shared, it is better to use the same functionality to do this so reduce bugs/maintainance etc